esp-hal/src/gpio/mod.rs

```diff
@@ -832,6 +842,17 @@ macro_rules! gpio {
             /// # Safety
             ///
             /// Ensure that only one instance of a pin is in use at one time.
+            ///
+            /// ## Example
+            ///
+            /// ```rust, no_run
+            #[doc = $crate::before_snippet!()]
+            /// use esp_hal::gpio::AnyPin;
+            /// let pin = unsafe { AnyPin::steal(1) };
```

Comment:
Let's reduce the unsafe here to only the required usage, we can safely create `AnyPin`s from GPIO4 like the other examples, then just showcase the clone part. Do the same for reborrow too please.